protobufs/livekit_agent.proto

```diff
@@ -21,154 +21,178 @@ option ruby_package = "LiveKit::Proto";
 
 import "livekit_models.proto";
 
-message WorkerInfo {
-    string id = 1;
-    string namespace = 2;
-    string version = 3;
-    string name = 4;
-    JobType type = 5;
-    ParticipantPermission allowed_permissions = 6;
+service AgentService {
+  rpc StartAgentJob(StartAgentJobRequest) returns (Job);
+  rpc StopAgentJob(StopAgentJobRequest) returns (Job);
+  rpc ListAgentJobs(ListAgentJobsRequest) returns (ListAgentJobsResponse);
 }
 
-message AgentInfo {
-    string id = 1;
-    string name = 2;
-    string version = 3;
+message StartAgentJobRequest {
+  JobType type = 1;
+  string room_name = 2;
+  optional string participant_identity = 3;
+  string namespace = 4;
+  string metadata = 5;
+}
```

Comment:
in the current implementation publisher jobs apply to all participants once they begin publishing. this doesn't fit into this api. multiple job ids could result from a single invocation and more jobs could start after the initial call.

i think we've accidentally overloaded the `Job` concept - in the original design this represented an an invocation of the agent worker. what we're creating with this api call is a rule for dispatching jobs ie start a job for the room or start a job for publishers or for one publisher with the identity x.

Reply:
maybe just `StartAgent` ?

Reply:
I think we have 2 options:
- Change the Job definition to represent 1 invocation for either the whole room, or 1 participant
- Keep the current Job definition and remove the `participant_identity` argument in the `Start` message (and rename the rpc indeed). 

Is there a use case for starting an Agent Job for a single participant? At first glance, it seems so in context where there is a main speaker. Short of doing that, we may eventually need some kind of filter argument to tell a 3rd party agent what participants to process.

Reply:
The current (before this PR) Job definition already has an optional `ParticipantInfo` field. How are agents currently expected to use it?

Reply:
we'll want the option to specify one or more publishers a job should start for. we don't want the user to have to specify the identities of every publisher in the room if they want the job to start automatically for publishers though. i think we need a new concept to describe what this api operates on - i don't have a great name for it but `WorkOrder` fits the job/worker metaphor.

the publisher's `ParticipantInfo` is stored in this field by the agents framework when dispatching a publisher type job.